### PR TITLE
10277 assign customer to imported txns

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: app/models/accounting/transaction.rb [hunk 1]
```diff
@@ -65,6 +66,7 @@ class Accounting::Transaction < ApplicationRecord
 
   belongs_to :account, inverse_of: :transactions, foreign_key: :accounting_account_id
   belongs_to :project, inverse_of: :transactions, foreign_key: :project_id
+  belongs_to :customer, foreign_key: :accounting_customer_id
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply app/models/accounting/transaction.rb]: this is in 10276

**Comment by DevneyHamilton** [comment]
There is no UI for this. It is attempting to address the DuplicateName errors on loans whose transactions have quickbooks customers with names different than the Madeline organization (e.g. Thread Coffee Roasters), but it may not be a complete solution. It assigns Accounting::Customers to transactions when they are imported from quickbooks based on quickbooks data.  I manually tested on sandbox by setting up a new loan, creating a disbursement in the sandbox qb company to go with the loan, with the customer set, imported by loading the txns for the loan in Madeline, and making sure the customer appeared in the read-only modal for the txn. For staging, we could try loans 
1003
1476
1477
2614
2987
2870
and see if they have new errors besides DuplicateName, or no errors. If not, I'll need to do more digging either on staging or with the setup Melody is working on.

**Comment by DevneyHamilton** [comment]
QA: this is a fix for a problem with Loan 1003. Right now, Loan 1003 shows a multiloan txn error and a "There was an error communicating with QuickBooks. An administrator has been notified. Some data may be out of date. (Error: Duplicate Name Exists Error: The name supplied already exists. : null)" error. If the changes in 10277 work, then Loan 1003 on staging will only show a multiloan transaction error.

**Review by meltheadorable — COMMENTED** [review]
Do we need to implement extractors for the customer for non-journal-entry transaction types? This otherwise looks great, and those can be separate tickets, just asking.

**Comment by DevneyHamilton** [comment]
> Do we need to implement extractors for the customer for non-journal-entry transaction types? This otherwise looks great, and those can be separate tickets, just asking.

In theory, yes, but I'm 99% sure it's not needed for existing data. I think it would be needed in the edge case that the only txns on a loan are bill, purchase, or deposit and then Madeline needs to make an interest txn on a loan for an org whose name doesn't match. This is a small enough edge case that I think it could go in low priority tickets?

**Comment by emitche** [comment]
@DevneyHamilton This looks fine in terms of tests passing and code being introduced.

Are there specific things you would like me to test for with the QA process? Should I test the customer assignments on a local environment?

Thank you!

**Comment by DevneyHamilton** [comment]
> @DevneyHamilton This looks fine in terms of tests passing and code being introduced.
> 
> Are there specific things you would like me to test for with the QA process? Should I test the customer assignments on a local environment?
> 
> Thank you!

QA: this is a fix for a problem with Loan 1003. Right now, Loan 1003 shows a multiloan txn error and a "There was an error communicating with QuickBooks. An administrator has been notified. Some data may be out of date. (Error: Duplicate Name Exists Error: The name supplied already exists. : null)" error. If the changes in 10277 work, then Loan 1003 on staging will only show a multiloan transaction error.

> There is no UI for this. It is attempting to address the DuplicateName errors on loans whose transactions have quickbooks customers with names different than the Madeline organization (e.g. Thread Coffee Roasters), but it may not be a complete solution. It assigns Accounting::Customers to transactions when they are imported from quickbooks based on quickbooks data. I manually tested on sandbox by setting up a new loan, creating a disbursement in the sandbox qb company to go with the loan, with the customer set, imported by loading the txns for the loan in Madeline, and making sure the customer appeared in the read-only modal for the txn. For staging, we could try loans
> 1003
> 1476
> 1477
> 2614
> 2987
> 2870
> and see if they have new errors besides DuplicateName, or no errors. If not, I'll need to do more digging either on staging or with the setup Melody is working on.

@emitche does this address QA questions? Above I described a manual test w/ sandbox I did while developing; you could try the same thing in sandbox for QA.

**Comment by DevneyHamilton** [comment]
> There is no UI for this. It is attempting to address the DuplicateName errors on loans whose transactions have quickbooks customers with names different than the Madeline organization (e.g. Thread Coffee Roasters), but it may not be a complete solution. It assigns Accounting::Customers to transactions when they are imported from quickbooks based on quickbooks data. I manually tested on sandbox by setting up a new loan, creating a disbursement in the sandbox qb company to go with the loan, with the customer set, imported by loading the txns for the loan in Madeline, and making sure the customer appeared in the read-only modal for the txn. For staging, we could try loans
> 1003
> 1476
> 1477
> 2614
> 2987
> 2870
> and see if they have new errors besides DuplicateName, or no errors. If not, I'll need to do more digging either on staging or with the setup Melody is working on.

@emitche does this address QA questions? I wasn't sure if you saw this or not (if not, I can @ you in the future). I described a manual test I did on sandbox, and you could try to duplicate it if you want to test before deploying to staging.

**Comment by emitche** [comment]
@DevneyHamilton I missed the earlier message. No need to @ me in the future. Sorry I missed the earlier message.

**Comment by emitche** [comment]
Resolves duplicate name errors on the following loans:
1003
1476
1477
2614
2987
2870
